boards/xtensa/esp32/Kconfig.board

```diff
@@ -0,0 +1,9 @@
+# Kconfig - ESP32 board configuration
+#
+# Copyright (c) 2017 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_ESP32
+	bool "ESP32 Development Board"
+	depends on XTENSA
+	select BOARD_XTENSA
```

Comment:
This is sort of done inside `xtensa_rtos.h` now: if `CONFIG_SIMULATOR_XTENSA` is defined, `XT_SIMULATOR` is defined; otherwise, `XT_BOARD` gets defined.  This `select` can be removed from `Kconfig.board`.